### PR TITLE
Add setting to turn extending numeric precision on or off

### PR DESCRIPTION
**Pull request by AMoo-Miki**
### Description

* Add setting to turn extending numeric precision on or off
* Fix a typo while inspecting values for large numerals in OSD and the JS client

## Screenshot
Screenshot 2024-11-08 at 16-58-25 OpenSearch Dashboards

## Changelog
- fix: Fix a typo while inspecting values for large numerals in OSD and the JS client
- feat: Add setting to turn extending numeric precision on or off

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [X] Update [CHANGELOG.md](./../CHANGELOG.md)
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/data/common/constants.ts [hunk 1]
```diff
@@ -108,4 +108,5 @@ export const UI_SETTINGS = {
   QUERY_DATAFRAME_HYDRATION_STRATEGY: 'query:dataframe:hydrationStrategy',
   SEARCH_QUERY_LANGUAGE_BLOCKLIST: 'search:queryLanguageBlocklist',
   NEW_HOME_PAGE: 'home:useNewHomePage',
+  DATA_WITH_LONG_NUMERALS: 'data:withLongNumerals',
```
Comment (kavilla) [inline src/plugins/data/common/constants.ts:111]: nit: i think i originally did `data` as well. but then was convinced to use `search` because then we can group it with the search category. not a strong opinion on this though

**Review by kavilla — DISMISSED** [review]

**Review by kavilla — APPROVED** [review]
sorry my pr caused a conflict on this pr so i just fixed it on github. do you mind verifying miki it's all good?

also one thing i think we should do is move this to the workspace setting or even better yet allow for request options from the query editor would make life a lot easier for developers vs an advanced setting to toggle percision or not

**Comment by AMoo-Miki** [comment]
> sorry my pr caused a conflict on this pr so i just fixed it on github. do you mind verifying miki it's all good?

Thanks. Didn't see anything glaring.

> also one thing i think we should do is move this to the workspace setting or even better yet allow for request options from the query editor would make life a lot easier for developers vs an advanced setting to toggle percision or not

Requests can choose to ignore or entertain the precision setting. I do see value for this to ALSO be in the workspace setting since not all users will have workspaces enabled. Will have to see how values are inherited from config and advanced settings to the workspace (later). Thanks for the suggestion.

File: src/plugins/console/public/application/hooks/use_send_current_request_to_opensearch/use_send_current_request_to_opensearch.ts [hunk 2]
```diff
@@ -112,5 +120,14 @@ export const useSendCurrentRequestToOpenSearch = (dataSourceId?: string) => {
...
-  }, [dispatch, http, dataSourceId, settings, notifications.toasts, trackUiMetric, history]);
+  }, [
+    dispatch,
+    http,
+    dataSourceId,
+    settings,
+    notifications.toasts,
+    trackUiMetric,
+    history,
+    uiSettings,
```
Comment (joshuali925) [inline src/plugins/console/public/application/hooks/use_send_current_request_to_opensearch/use_send_current_request_to_opensearch.ts:131]: uiSettings is a service, would it ever change?

File: src/plugins/data/server/ui_settings.ts [hunk 3]
```diff
@@ -535,6 +535,17 @@ export function getUiSettings(
...
       schema: schema.string(),
     },
+    [UI_SETTINGS.DATA_WITH_LONG_NUMERALS]: {
+      name: i18n.translate('data.advancedSettings.data.withLongNumeralsTitle', {
+        defaultMessage: 'Extend Numeric Precision',
+      }),
+      value: true,
+      description: i18n.translate('data.advancedSettings.data.withLongNumeralsText', {
+        defaultMessage:
+          "Turn on for precise handling of extremely large numbers. Turn off to optimize performance when high precision for large values isn't required.",
```
Comment (joshuali925) [inline src/plugins/data/server/ui_settings.ts:545]: should this tell user what the threshold is before losing precision?

**Review by joshuali925 — APPROVED** [review]
nothing blocking

File: src/plugins/console/public/application/hooks/use_send_current_request_to_opensearch/use_send_current_request_to_opensearch.ts (hunk 2 above)
Comment (AMoo-Miki) [reply src/plugins/console/public/application/hooks/use_send_current_request_to_opensearch/use_send_current_request_to_opensearch.ts:131]: Most of these are; not sure if it is because `useServicesContext` can throw (which would prevent code advancing here), linting rules, or something else but most of the services fetched from the context are passed in here too. Didn't want to re-engineer this section so followed the same pattern.

File: src/plugins/data/server/ui_settings.ts (hunk 3 above)
Comment (AMoo-Miki) [reply src/plugins/data/server/ui_settings.ts:545]: It would be too wordy. Instead, I will have this added to the documentation website with proper explanation.

**Review by kavilla — APPROVED** [review]

**Review by virajsanghvi — APPROVED** [review]
Code lgtm, although does look like there are some failures?

**Comment by AMoo-Miki** [comment]
@virajsanghvi thanks; just flaky test.
